main.js

```diff
@@ -1,16 +1,57 @@
 define(function (require, exports, module) {
 	var fileInfo = {};
 
+	// Load preferences
+	var PreferencesManager = brackets.getModule("preferences/PreferencesManager"),
+		prefs = PreferencesManager.getExtensionPrefs("file-icons");
+	prefs.definePreference("icons", "array", []);
+
+	// Declare variables
+	var iconArray;
```

Comment:
Call this variable `userIcons`, place it directly under `var fileInfo = {}` and rename `fileInfo` to `defaultIcons`